src/Core/Domain/CartRule/Command/SetCartRuleRestrictionsCommand.php

```diff
@@ -59,6 +65,23 @@ public function __construct(
         $this->cartRuleId = new CartRuleId($cartRuleId);
     }
 
+    public function setRestrictedCarrierIds(array $restrictedCarrierIds): self
+    {
+        $this->restrictedCarrierIds = array_map(static function ($carrierId): CarrierId {
+            return new CarrierId($carrierId);
+        }, $restrictedCarrierIds);
```

Comment:
```suggestion
        $this->restrictedCarrierIds = array_map(static fn ($carrierId): CarrierId => new CarrierId($carrierId), $restrictedCarrierIds);
```
Still not blocking 👍

Reply:
@lartist agreed, I think we can merge the two first PRs, and apply those optimizations everywhere in the last one